src/ApiCore/Version.php

```diff
@@ -42,12 +42,17 @@ public static function getApiCoreVersion()
     {
         if (is_null(self::$version)) {
             $versionFile = implode(DIRECTORY_SEPARATOR, [__DIR__, '..', '..', 'VERSION']);
-            $versionString = @file_get_contents($versionFile) ?: "";
-            self::$version = trim($versionString);
+            self::$version = self::readVersionFile($versionFile);
         }
         return self::$version;
     }
 
+    public static function readVersionFile($file)
+    {
+        $versionString = @file_get_contents($file) ?: "";
+        return trim($versionString);
```

Comment:
If for some reason there's no `VERSION` file, it will return an empty string.

Are you sure that an empty string will work without problem?
How about to return "Unknown-Version" or sth?

Reply:
Hmm, I think I prefer empty string rather than some placeholder. We want to keep the possibility of testing for the version being missing. What about returning null? That has disadvantages as well (it is even more likely to not work somewhere).

Another alternative - have the function take a second argument, $defaultValue, that is the value to use if no version file is found. WDYT?

Reply:
Alright, an empty string works for me.